app/models/page_view.rb

```diff
@@ -5,6 +5,7 @@ class PageView < ApplicationRecord
   belongs_to :article
 
   before_create :extract_domain_and_path
+  after_create :adjust_tweet_stat_on_article
```

Comment:
This will potentially create a lot of jobs for popular articles. Yes, because of the logic you put in the service not all of them will result in calling the Twitter API but this will literally create a job *we* have to consume for each page view

Reply:
It kind of trade-off here. Either many jobs (in that queue), or calling `MarkdownParser#liquid_nodes` (calling `Liquid::Template.parse`) every time a `PageView` is created. I believe calling `Liquid::Template.parse` is more expensive, then queuing jobs. Especially if the queue is set to lowest priority..

Reply:
I think we should call this only every `n` pageviews, so something like `if rand(n) == 1` (`n` being something like 50) kind of thing. Not something we need to check for every time a pageview is created. If we find we want refresh this more often, we can lower what `n` is